### ISSUE TITLE
Implement Trac bridge

### ISSUE THREAD

**Issue by strk** [MichaelMure/git-bug#237]
For those of us resisting dropping old Trac in favour of other non-distributed solutions.

See https://trac.edgewall.org/

**Comment by MichaelMure** [comment]
Yep that would be nice. It's quite far on the current roadmap though. 

However, it's a great area to start contributing if you are willing. During the summer, @A-Hilaly improved the bridge core and developed the advanced features like clean cancel or this kind of things. Developing a new bridge would be mimicking what is done with Github and Gitlab but with another API.

**Comment by A-Hilaly** [comment]
Agreed adding new bridges won't be a big deal.  I can't find the official documentation for https://trac.edgewall.org/ API. @strk any idea where i can find that ?

**Comment by strk** [comment]
There is no native API for Trac, but an XMLRPC plugin exists:
https://trac-hacks.org/wiki/XmlRpcPlugin

The xmlrpc endpoint seems to be used by the [tracboat](https://github.com/tracboat/tracboat) which implements Trac to GitLab migration.

**Comment by strk** [comment]
Commandline tool for Trac also uses XmlRpc plugin: https://code.google.com/archive/p/tracshell/

**Comment by strk** [comment]
Another alternative would be hooking on commandline tool that can already interact with trac, like I did to integrate Trac with Mutt: https://strk.kbt.io/blog/2015/11/11/trac-from-mutt/

That tool doesn't require XmlRpc plugin to be installed on the Trac server.
